src/ansys/geometry/core/tools/problem_areas.py

```diff
@@ -506,3 +507,46 @@ def fix(self) -> RepairToolMessage:
             response.result.modified_bodies_monikers,
         )
         return message
+
+
+class UnsimplifiedFaceProblemAreas(ProblemArea):
+    """Represents a unsimplified face problem area with unique identifier and associated faces."""
+
+    def __init__(self, id: str, grpc_client: GrpcClient, faces: list["Face"]):
+        """Initialize a new instance of the unsimplified face problem area class."""
+        super().__init__(id, grpc_client)
+
+        from ansys.geometry.core.designer.face import Face
+
+        # Verify that all elements in the list are of type Body
+        check_type_all_elements_in_iterable(faces, Face)
+
```

Comment:
I am not sure whether it is necessary to do this check here... Users won't be creating these problem areas themselves. They will be getting them from the tools results. Meaning that we have control over the inputs. I would avoid the check in this constructor